components/Editor.js

```diff
@@ -172,6 +174,14 @@ class Editor extends React.Component {
                   .replace(/%23/g, '#')
                   .replace(/%0A/g, '\n')
               )
+              // remove other fonts which are not used
+              .then(
+                dataUrl =>
+                  (dataUrl = dataUrl.replace(
```

Comment:
Is it possible to use the `dom-to-image`'s `filter` option (https://github.com/dawnlabs/carbon/pull/683/files#diff-24387bc9bda6a21543364a64ae3b83c1R152) in order to remove unwanted fonts? If you can even separate them into separate `style` nodes if that make's it easier

Reply:
Thank you @mfix22 for your comments and suggestions. It's a cool suggestion, and I have never thought of this.

The issues that I faced on implementing the suggestion -

1. I tried to fix it with the `filter`. But the styles nodes (having font-face) are appended by the `embedFonts` in `dom-to-image` (`embedFonts` having the `newFontFaces` in `dom-to-image` reads all the `document.styleSheets` to get the font-faces.), Thus the `filter` did not come across the style nodes, so was not able to use `filter` to prevent unwanted style nodes.
2. And there is no option to update the embedded fonts in `dom-to-image`.

This issue to **Optimize downloading webfonts** are being discussed [https://github.com/tsayen/dom-to-image/issues/37](here) in the [https://github.com/tsayen/dom-to-image](dom-to-image)

We can make this suggestion happen by updating the https://github.com/tsayen/dom-to-image/blob/master/src/dom-to-image.js to get only the fonts needed.

Reply:
Please correct me if any thing that i mentioned above is wrong.

Reply:
👍